### PR TITLE
tools: allow test runner to target different test server

### PR DESCRIPTION
**Pull request by LPardue**
Small change to `runner.rs` to allow it agility in what server it actually targets for tests. It decouples the server SNI (used by QUIC) from the IP and port used by UDP socket. 

The httpbin_tests.rs are updated with a new environment variable `HTTPBIN_HOST` to make this easy to use from a testing environment.

### CODE REVIEW

File: tools/http3_test/src/runner.rs [hunk 1]
```diff
@@ -28,7 +28,7 @@ use std::net::ToSocketAddrs;
 
 use ring::rand::*;
 
-pub fn run(test: &mut crate::Http3Test, verify_peer: bool) {
+pub fn run(test: &mut crate::Http3Test, host: &str, verify_peer: bool) {
```
Comment (ghedo) [inline tools/http3_test/src/runner.rs]: Instead of a string, maybe pass a `SocketAddr` directly? https://doc.rust-lang.org/std/net/enum.SocketAddr.html

File: tools/http3_test/tests/httpbin_tests.rs [hunk 2]
```diff
@@ -25,6 +25,16 @@ mod httpbin_tests {
         url
     }
 
+    fn host() -> String {
+        let host = match std::env::var_os("HTTPBIN_HOST") {
+            Some(val) => val.into_string().unwrap(),
+
+            None => String::from("cloudflare-quic.com:443"),
```
Comment (ghedo) [inline tools/http3_test/tests/httpbin_tests.rs]: Do we always want users to set this when running the tests? An alternative would be to use `HTTPBIN_ENDPOINT` host by default, unless `HTTPBIN_HOST` is set. This way we also wouldn't need to hard-code cloudflare-quic.com in two places.

For example, `host()` could take the test URL as parameter, and return the value from that if `HTTPBIN_HOST` is not set.

File: tools/http3_test/src/runner.rs (hunk 1 above)
Comment (LPardue) [reply tools/http3_test/src/runner.rs]: I did that first but then realised it's a bit annoying if you just want to pass a string directly into runner.rs. 

It's a pretty close call on which is better.

File: tools/http3_test/tests/httpbin_tests.rs (hunk 2 above)
Comment (LPardue) [reply tools/http3_test/tests/httpbin_tests.rs]: Makes sense, ta.

File: tools/http3_test/src/runner.rs (hunk 1 above)
Comment (LPardue) [reply tools/http3_test/src/runner.rs]: It was actually easier to handle some edge cases (related to your other comment) by only accepting a SocketAddr

File: tools/http3_test/tests/httpbin_tests.rs (hunk 2 above)
Comment (LPardue) [reply tools/http3_test/tests/httpbin_tests.rs]: done

**Comment by ghedo** [comment]
I tweaked the commit message slightly when merging.
